pkg/log/logcontext_test.go

```diff
@@ -0,0 +1,189 @@
+package log
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/mocks"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/runtime/security"
+
+	"github.com/openshift/assisted-service/client"
+	clientInstaller "github.com/openshift/assisted-service/client/installer"
+	"github.com/openshift/assisted-service/restapi"
+	"github.com/openshift/assisted-service/restapi/operations/installer"
+)
+
+func serv(server *http.Server) {
+	_ = server.ListenAndServe()
+}
+
+type AuthPayload struct {
+	Username string `json:"username"`
+	IsUser   bool   `json:"is_user"`
+}
+
+var testUser = AuthPayload{
+	IsUser:   true,
+	Username: "dummy",
+}
+
+var authenticator = func(name string, _ string, authenticate security.TokenAuthentication) runtime.Authenticator {
+	return security.HttpAuthenticator(func(r *http.Request) (bool, interface{}, error) {
+		return true, &testUser, nil
+	})
+}
+
+func createServer(logger *logrus.Logger, installer *restapi.InstallerAPI) *http.Server {
+	h, _ := restapi.Handler(restapi.Config{
+		AuthAgentAuth:       nil,
+		AuthUserAuth:        nil,
+		APIKeyAuthenticator: authenticator,
+		Authorizer:          func(*http.Request) error { return nil },
+		InstallerAPI:        *installer,
+		EventsAPI:           nil,
+		Logger:              logger.Printf,
+		VersionsAPI:         nil,
+		ManagedDomainsAPI:   nil,
+		InnerMiddleware:     ContextHandler(),
+	})
+
+	server := &http.Server{Addr: "localhost:8082", Handler: h}
+	go serv(server)
+	return server
+}
+
+func createClient() *client.AssistedInstall {
+	cfg := client.Config{
+		URL: &url.URL{
+			Scheme: client.DefaultSchemes[0],
+			Host:   "localhost:8082",
+			Path:   client.DefaultBasePath,
+		},
+	}
+
+	return client.New(cfg)
+}
+
+func waitForServer(bmclient *client.AssistedInstall, mockInstallApi *mocks.MockInstallerAPI) {
+	var err error = fmt.Errorf("start polling server...")
+	//loop up to a second to wait for the server to go up
+	mockInstallApi.EXPECT().ListClusters(gomock.Any(), gomock.Any()).DoAndReturn(
+		func(ctx context.Context, params installer.ListClustersParams) middleware.Responder {
+			return installer.NewListClustersOK()
+		}).AnyTimes()
+	var i int
```

Comment:
you can probably replace all this block with https://godoc.org/github.com/onsi/gomega#Eventually

Reply:
see my reply offline